src/config/analyticsRoutes.ts

```diff
@@ -0,0 +1,271 @@
+    "loaded screen: confirm mnemonic phrase" as AnalyticsEvent,
+  [AUTH_STACK_ROUTES.IMPORT_WALLET_SCREEN]:
+    "loaded screen: import account" as AnalyticsEvent,
+  [ROOT_NAVIGATOR_ROUTES.LOCK_SCREEN]:
+    "loaded screen: unlock account" as AnalyticsEvent,
+  [MAIN_TAB_ROUTES.TAB_HOME]: "loaded screen: account" as AnalyticsEvent,
+  [MAIN_TAB_ROUTES.TAB_HISTORY]:
+    "loaded screen: account history" as AnalyticsEvent,
+  [MAIN_TAB_ROUTES.TAB_DISCOVERY]: "loaded screen: discover" as AnalyticsEvent,
+  [ROOT_NAVIGATOR_ROUTES.ACCOUNT_QR_CODE_SCREEN]:
+    "loaded screen: view public key generator" as AnalyticsEvent,
+  [ROOT_NAVIGATOR_ROUTES.TOKEN_DETAILS_SCREEN]:
+    "loaded screen: asset detail" as AnalyticsEvent,
+  [SEND_PAYMENT_ROUTES.SEND_SEARCH_CONTACTS_SCREEN]:
+    "loaded screen: send payment to" as AnalyticsEvent,
+  [SEND_PAYMENT_ROUTES.TRANSACTION_AMOUNT_SCREEN]:
+    "loaded screen: send payment amount" as AnalyticsEvent,
+  [SEND_PAYMENT_ROUTES.TRANSACTION_MEMO_SCREEN]:
+    "loaded screen: send payment settings" as AnalyticsEvent,
+  [SEND_PAYMENT_ROUTES.TRANSACTION_FEE_SCREEN]:
+    "loaded screen: send payment fee" as AnalyticsEvent,
+  [SEND_PAYMENT_ROUTES.TRANSACTION_TIMEOUT_SCREEN]:
+    "loaded screen: send payment timeout" as AnalyticsEvent,
+  [SETTINGS_ROUTES.CHANGE_NETWORK_SCREEN]:
+    "loaded screen: manage network" as AnalyticsEvent,
+  [SETTINGS_ROUTES.NETWORK_SETTINGS_SCREEN]:
+    "loaded screen: network settings" as AnalyticsEvent,
+  [SETTINGS_ROUTES.SHARE_FEEDBACK_SCREEN]:
+    "loaded screen: leave feedback" as AnalyticsEvent,
+  [SETTINGS_ROUTES.SHOW_RECOVERY_PHRASE_SCREEN]:
+    "loaded screen: show recovery phrase" as AnalyticsEvent,
+  [BUY_XLM_ROUTES.BUY_XLM_SCREEN]: "loaded screen: add fund" as AnalyticsEvent,
+};
+
+/**
+ * Safely checks if a string is valid and non-empty.
+ */
+const isValidString = (value: unknown): value is string =>
+  typeof value === "string" && value.trim().length > 0;
+
+/**
+ * Transform route name to analytics event name automatically.
+ * Examples:
+ * - "WelcomeScreen" → "loaded screen: welcome"
+ * - "SettingsScreen" → "loaded screen: settings"
+ * - "SwapAmountScreen" → "loaded screen: swap amount"
+ */
+const transformRouteToEventName = (routeName: string): string => {
+  // Remove "Screen" suffix if present
+  const baseName = routeName.replace(/Screen$/, "");
+
+  // Convert PascalCase to lowercase with spaces
+  // "SwapAmount" → "swap amount"
+  const withSpaces = baseName
+    .replace(/([A-Z])/g, " $1") // Add space before capitals
+    .toLowerCase()
+    .trim();
+
+  return `loaded screen: ${withSpaces}`;
+};
+
+/**
+ * Gets all route objects to scan for analytics mapping.
+ */
+const getAllRouteObjects = () =>
+  [
+    ROOT_NAVIGATOR_ROUTES,
+    AUTH_STACK_ROUTES,
+    MAIN_TAB_ROUTES,
+    MANAGE_ASSETS_ROUTES,
+    SETTINGS_ROUTES,
+    MANAGE_WALLETS_ROUTES,
+    BUY_XLM_ROUTES,
+    SEND_PAYMENT_ROUTES,
+    SWAP_ROUTES,
+  ] as const;
```

Comment:
would be nice if we could find a way of raising a lint error here or something that would let devs know in case some route object is missing here. I'm thinking about the future when we start creating new routes, we'd need to remember to add them here as well. Wdyt?